### DISCUSSION TITLE
Golang - When to use SDK vs Library

### DISCUSSION THREAD

**Discussion by patmacpherson-sc** [open-policy-agent/community#232] [OPA and Rego]
Hello, firstly - thanks for the great work with OPA!

I'm reading through this page - https://www.openpolicyagent.org/docs/latest/integration/

I'm working with Golang services, but am unsure whether I should use the API library or the SDK.

From what I can see, the SDK has a dependency on the API library, and the SDK offers a simpler experience. The API library looks to be the more popular.

If I'm working in Go services, how do I choose between the two?

**Comment by anderseknert** [comment]
Hi there! Unless doing something specialized, I’d go with the SDK. This provides you all the capabilities of OPA out of the box, and you (and/or your users) can simply drop in an OPA config file to use them. Having the same config as the OPA server simplifies things a lot, and allows someone to test their stuff on a local OPA, and bring that same config to your app later.
